co-authors-plus.php

```diff
@@ -1052,22 +1052,59 @@ function filter_count_user_posts( $count, $user_id ) {
 
 		if ( is_object( $user ) ) {
 			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
-
 			$term = $this->get_author_term( $user );
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
+			if ( ! $user->linked_account ) {
```

Comment:
Sometimes users created before CAP is installed will not have a proper term assigned and thus, may return `false`. We need to practice defensive coding for these situations. See https://github.com/Automattic/Co-Authors-Plus/pull/776#issuecomment-816812351 for further details.

Reply:
Thanks for pointing this out, @rebeccahum. I just addressed this problem by checking if `$term` exists for the current user.

Reply:
@nielslange Is there anywhere else you could see that could use some defensive coding as well?

Reply:
I've updated the logic in the following way:

- If a `$user->linked_account` exists, return the count of the linked account using `get_linked_post_count()`.
- If the `$term->count` is not null, return the `$term->count`.
- If none of the two criteria above are true, return the unfiltered `$count`.